### PR TITLE
tests(cubicspline): add tests

### PR DESCRIPTION
**Pull request by nritsche**
`test_polynomial_edge` currently fails, but it probably shouldn't.

### CODE REVIEW

File: tests/test_cubicspline.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+
+@pytest.mark.xfail
+def test_polynomial_edge():
+    """
+    Interpolate f(x) = 1 + 2x + 3x**2.
+
+    Tests values close to the lower edge of the interpolation function.
+
+    This test currently fails.
```
Comment (anjakefala) [inline tests/test_cubicspline.py:113]: Fails on James' branch? Or is this test not part of the build process, and that is why the build is green.

Comment (anjakefala) [reply tests/test_cubicspline.py:113]: Should we open an issue to investigate the test failure?

**Comment by anjakefala** [comment]
Update: I now understand, and noticed the `xfail`. I only have one question then:

are we content to take on fixing that test as a future thing, as opposed to a thing we fix in the present?

**Comment by jrs65** [comment]
I've just had a look at comparing `cora` and `scipy.interpolate`. As suspected all the differences come from the different boundary conditions being enforced:
```python
import numpy as np
import scipy.interpolate as ci
from cora.util import cubicspline as cs

def f(x):
    return x**3 * np.arcsinh(x**4 - 7*x)

x_fine = np.linspace(-10, 10, 10000)
x_coarse = np.linspace(-10, 10, 70)

f_si1 = ci.CubicSpline(x_coarse, f(x_coarse), bc_type="not-a-knot")  # default
f_si2 = ci.CubicSpline(x_coarse, f(x_coarse), bc_type="natural")  # same type as cora
f_cs = cs.Interpolater(x_coarse, f(x_coarse))

fig, axes = plt.subplots(2, 1)
axes[0].plot(x_fine, f(x_fine), label="True")
axes[0].plot(x_coarse, f(x_coarse), "rx", label="Interpolation points")
axes[0].plot(x_fine, f_si1(x_fine), label="Scipy not-a-knot")
axes[0].plot(x_fine, f_si2(x_fine), label="Scipy natural")
axes[0].plot(x_fine, f_cs(x_fine), label="Cora")
axes[0].legend()

axes[1].plot(x_fine, f_si1(x_fine) - f(x_fine), label="Scipy not-a-knot")
axes[1].plot(x_fine, f_si2(x_fine) - f(x_fine), label="Scipy natural")
axes[1].plot(x_fine, f_cs(x_fine) - f(x_fine), label="Cora")
axes[1].legend()

axes[1].set_xlabel("$x$")
axes[0].set_ylabel("$f(x)$")
axes[1].set_ylabel("Error in $f(x)$")
```
which gives this plot:
image

If I check the differences between them numerically they are extremely small with the same boundary conditions set:
```
>>> (f_cs(x_fine) - f_si1(x_fine)).std()
0.4031668816044425
>>> (f_cs(x_fine) - f_si2(x_fine)).std()
4.583647355531852e-13
```

**Comment by jrs65** [comment]
So in summary, I'm less worried about these issues than I was 20 mins ago. Although we may want to implement the "not-a-knot" criteria as an option as it seems to be pretty good.

**Review by anjakefala — APPROVED** [review]

**Review by jrs65 — APPROVED** [review]

**Pull request merged**